### DISCUSSION TITLE
기초프로젝트 2주차 팀회고 (최종)

### DISCUSSION THREAD

**Discussion by yeji0512** [codeit-bootcamp-frontend/2-Weekly-Mission#314] [회고]
기초 프로젝트를 무사히 마무리 하신 여러분,
너무 너무 고생 많으셨습니다 💜👏

2주간의 팀 프로젝트를 돌아보며 서로 고마웠던 점을 나누어보고, 속상했던 것이 있다면 함께 풀 수 있는 시간이 되었으면 좋겠어요. 
저번과 같이 KPT 회고로 진행해주신 후, 다음주 화요일(12/26)까지 댓글 남겨주세요.

**Comment by jisunipark** [comment]
# 8팀

## Keep

- 빠지는 부분 없이 **기본적 구현**을 잘 하였다.
    - 스크롤 부분 구현이 잘 되었다. (준석님👍)
- **단순**해서 그런지 로딩이 빠르다.
    - 이렇게 렌더링을 최소하는 방법이 좋은 것 같다. **렌더링 최적화**!
    - post/id 페이지 코드를 단순하게 잘 짠 것 같다. (필겸님👍)
- 페이지를 합쳐지는 부분이 빠르고 원활하게 해결이 잘 되었다.
    - 페이지 별로 역할을 분담했기 때문
- 모달창 작업에서 필겸님의 코드를 참고하셔서 작업하신 성연님(👍)에게서 ****짬바를..

## Problem

- **통일성**의 문제
    - 같은 기능을 가진 컴포넌트 통일성 문제
        - 페이지 별로 역할을 분담했기 때문
    - 공통 요소 스타일 통일성 문제
    - 네이밍 컨벤션에 대한 부분이 실천이 잘 안 됨
- api 명세 이해 부족으로 인해 막판에 잘 안 되었던 부분 (필겸) → 커뮤니케이션의 문제
- 체계적이지 못했던 **PR 머지 시스템**과 **PR 리뷰**의 부족
    - PR 머지 단위가 너무 컸음. 한 페이지 통으로 PR 올리는 등.
    - 머지가 제때 되지 않아 커밋이 중첩되는 문제
    - 다른 팀원의 코드에 대한 이해 부족 → 다른 분들을 도와주려고 해도 도와줄 수가 없는 문제

## Try

- 통일성의 문제
    - 좀 더 자주 소통 하기
    - 공통된 것을 마주했을 때 미루지 않고 바로 해결하기
- PR 시스템 문제
    - GitHub의 기능을 좀 더 활용해보기 (PR 태그, back log 기능)
    - PR을 **더 작은 단위**로, 자주 올리기
    - 팀원들의 코드에 **PR 리뷰** 적극적으로 달기
- typescript의 중요성을 느낌 (성연)
    - typescript를 쓰면 코드에서 어떤 prop을 사용하는지 보여서 편함

**Comment by eonpain** [comment]
## 9️⃣ 팀

### ⭐️ Keep
- 각자 맡았던 파트를 완수하더라도 즉각적인 소통으로 타 팀원들의 파트를 돕거나 새로운 파트를 맡아 협업의 효율과 속도를 높임.
- 프로젝트의 처음부터 끝나는 순간까지 프로젝트의 모든 과정들을 기록, 정리를 성실히 함.
- 각자 다른 기능을 맡아 합치더라도 충돌 문제가 생기지 않도록 페이지 단위로 효율적인 R&R 분배를 함.
- 사용자가 보기 편한 디자인 테마를 적용함으로써 미적 요소를 반영함.
- 모든 기본적인 기능을 성공적으로 구현함으로서 코딩에 대한 자신감이 생김.

### 🤢 Problem
- 한파, 늦은 시간까지의 작업으로 인한 컨디션 난조로 스크럼 시간이 지연됨.
- 일부 반응형 페이지 구성에서 시간을 많이 소요함. 
- 추가 기능을 거의 시도하지 못한 점이 아쉽다. (특히 삭제하기 기능에 대한 비밀번호 설정기능을 추가하지 못한 것)
- PR을 남기고 스스로 스쿼시 머지를 해서 검토 없이 기능들이 합쳐진 경우가 존재했다.

### 🏃🏻 Try
- 디스코드 화면 공유를 통해 각자 작업한 기능들을 설명, 공유해 타 코드를 이해하는 시간을 절약함.
- 멘토링 날짜를 조정해 프로젝트 내 문제에 대한 조언을 얻어 수월히 문제해결을 함. 
- PR 후 머지를 받을 수 있는 조건들을 팀원들과 만들어야 할 필요성을 느낌.
- 추후 리팩토링을 통해 추가기능을 만들 것이다.

![image](https://media.tenor.com/P0EUlHajOeEAAAAi/%EC%9D%B4%EC%84%B8%EB%8F%8C-%EA%B3%A0%EC%84%B8%EA%B5%AC.gif)

**Comment by hankim0904** [comment]
# 🍀4팀

## Keep

- 매일 팀미팅 시간에 전날 한 일을 요약해서 팀원들에게 공유하고, 화면 공유를 통해 구현한 코드의 작동 여부를 보여줬음
- 매일 본인이 해야할 일을 정리해서 팀원들에게 알리고 지키기 위해 노력했음
- 모르는 내용이 있으면 즉시 팀원에게 도움을 구했음
- 팀원이 도움을 청하면 다른 팀원들은 함께 의논하고 해결책을 찾기 위해 힘썼음

## Problem

- 클래스 네이밍이 맞지 않음
- 기본 구조에 대해 이해도가 부족하여 미리 설정된 constants의 수정이 빈번하게 일어남
- Props Drilling 문제로 context API 리팩토링 필요
- PR 업로드 시 상세커밋 내용 부족
- 시간 부족으로 코드 리뷰를 충분히 진행하지 못 했음

## try

- 일정 계획을 보다 구체적으로 세울 것
- 깃헙, 노션 등 기록 플랫폼에 담당자를 지정하여 보다 관리를 철저히 할 것
- 클래스 네임, 변수명 등 네이밍 규칙을 구체적으로 설정 할 것
- 데이터와 관련한 코드를 정확히 파악해서 효율적인 구조를 미리 파악하고 프로젝트를 시작할 것

**Comment by ys1008p** [comment]
# 7팀

## keep
 - 팀원간의 의사소통이 원활하고 적극적으로 참여하는 분위기를 유지했다.
 - 발표 직전까지도 최대한 프로젝트의 완성도를 위해 서로 노력하는 모습을 보여줬다.
 - 작업 중 모르는 게 있으면 적극적으로 의견를 교류하고 함께 해결책을 찾아본다.


## problem
 - 작업 중 우선순위를 정확히 설정하지 못하여 시간 분배에 어려움을 겪었다.
 - 노션이나 깃허브 같은 협업도구를 적극적으로 활용하지 못하였다.
 - 작업 진행 중에 코드리뷰가 원활하게 이루어지지 않았다.
 - 팀에서 정한 코드 컨벤션을 사용하기로 했는데 다들 익숙치 않고 정확한 활용법을 잘 몰라서 코드를 작성하는데 어려움을 겪었다.
 - git branch 사용시 적용하기로 한 컨벤션이 잘 지켜지지 않았다.


## Try
 - 다음 프로젝트에서는 협업도구를 이용한 협업방식을 확실하게 정해놓고 작업을 진행하면 좋을 것 같다.
 - main에 merge 전에 본인이 올린 PR코드를 팀원들과 공유 및 상의 후 merge하면 좋을 것 같다.
 - 사용하기로 결정한 컨벤션이 있다면 최대한 지키도록 노력하기.
 - 사용률이 높은 SCSS를 선택했는데 다음에는 react와의 호환성을 고려해서 styledcomponent를 사용해보기

**Comment by muyahho** [comment]
3팀 최종회고
![IMG_8185](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/88885994/67395301-9c73-4fc6-b148-2d6332b5f1b1)

**Comment by grapefruit13** [comment]
# 2팀

<img width="668" alt="image" src="https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/92169354/07ea5230-4555-4157-adf2-7c13e3258fe5">

**Comment by dkile** [comment]
# 11팀

## Keep
- 코드 리뷰: 보다 책임감있는 코드 작성
- 막판 리팩터링: 파일명, 폴더 구조 정리
- 현업의 협업 방식과 가장 유사한 방식으로 프로젝트를 진행해볼 수 있어서 좋았다! (+ 주강사님의 코멘트)

## Problem
- pr 승인의 기준이 모호, 작업 지연
- 미디어쿼리 브레이크포인트 논의를 미리 해두지 않아서 모두 기준이 제각각이었다.
- 프로젝트 초반과 후반의 소통 속도 차이
- 일정을 제대로 지키지 못 한 적이 많았다.

## Try
- 공통 컴포넌트 활용에 대한 논의를 코딩 이전에 해보기
- 브레이크포인트 미리 정해두기
- 질문을 두려워하지 말기
- 요구사항 분석 능력 키우기
- 일정울 크게 한 번 나누고 각 단위의 목표가 있으면 좋을 것 같다.

**Comment by rladngus133** [comment]
# 💩1팀

# Keep (좋았던 점)
- 다들 빠른 소통을 유지해주었음
- 팀 활동시간에 다들 빠지지않고 잘 참여해주었음
- 아이디어를 불편함없이 계속 말해주어 팀활동에 활기가 있었음
- 문제발생시 소통이 잘 되어 빠른 문제해결이 가능했음
- 이번 프로젝트로 협업에 대한 기본을 알게된것 같아 자신감이 생겼음

# Problem (문제)
- PR리뷰가 부족하였음
- 초기셋팅 단계에서 사용할 라이브러리 선택을 너무 가볍게 했음

# Try (문제해결)
- 셀프코드리뷰를 통한 꼼꼼한 PR체크 
- 초기셋팅시 프로젝트의 방향성을 잡고 확실한 계획설계

**Comment by bjiwon324** [comment]
# 12팀

## Keep
팀 내 소통을 위해 모든 팀원이 위워크에 매일 나왔다.

- 궁금한 점이 생기면 모두가 해결하기 위해 많은 분들에게 자문을 구했다...?

매일 미팅을 통해 각자의 진행 상황을 잘 공유한다.

매일 진행 상황을 팀 메세지를 통해 정리 및 공유하여 현재 위치를 파악할 수 있도록 했다.

각자 작업 방식이나, 작업 속도 등에 대한 다양성을 이해해줬다.

아이디어 제안을 적극적으로 한다.

제안된 아이디어를 긍정적으로 받아들이고 수렴하려 했다.

## Problem

깃을 다루는게 익숙하지 않아 충돌이나 에러가 종종 생겼다.
자신있게 프로젝트를 이끌 수 있는 역량은 부족했다.

본인 실력에 대한 자신감이 없다.
본인의 작업 속도를 정확하게 파악하지 못해 예상 작업 시간과 실제 작업 시간이 차이가 날 때가 있었다.

시간에 쫒기다보니 검수가 부족했다.(리팩토링 등 모두가 모든 내용을 이해하진 못했던 것 같다)



## Try
스스로의 실력에 대한 자신감을 가지자.
변수, 함수 네이밍을 서로가 이해하기 쉽게 직관적으로 고쳐보자.
커밋 메세지를 좀 더 이해하기 쉽게 작성하자.
다음 프로젝트부터는 코드 리뷰를 활성화 하자.

**Comment by 9ummy9** [comment]
# ✨10팀✨
 ### [Keep]
근아)
- 새로 정한 컨셉에 적극적으로 아이디어를 내고, 일찍 끝난 사람들이 다른 사람들의 일을 나눠맡는 자세가 좋았다

우진)
- 기능 구현을 하며 모르거나 애매했던 부분에 대한 의견을 나누고 몰랐던 새로운 부분을 알아갈 수 있어 좋았다.

성원) 
- 각자 맡은 파트를 끝까지 책임감을 가지고 작업을 끝마치는 부분이 좋았다

현기)
- 작업을 하다 보면 의논해야 할 사항이나 질문이 생기는데 코어 타임 이외의 시간에도 피드백이 빨라서 좋았다.
- 누가 시키지 않아도 스스로 수정할 사항을 찾아 개선하는 부분이 좋았다.
- 다들 열심히 노력하는 자세가 좋았다.

채연)
- 팀원들이 적극적으로 아이디어를 내서 재미있는 컨셉의 프로젝트가 완성된 것 같아 좋았다.
- 혼자라면 오래 걸렸을 일을 같이 머리를 맞대 상대적으로 빠른 시간내에 해결하는 경험을 해 좋았다.

 ### [Problem]
 근아)
- 발표 시간 10분을 넘기지 않으려고 팀규칙 소개등을 발표에서 제외한 것이 아쉽다.
- 디자인 수정 시 global css 값을 변경했는데 각 컴포넌트의 사용값을 명확히 파악하지 못해서 팀원들이 적용해보면서 수정해야했다.

우진)
- git을 통한 코드리뷰를 생각하고 프로젝트를 진행하다보니 시간적 여유가 없어 잘 다루지 못한거같아 아쉽다.

성원)
- 컨셉에 맞춘 디자인변경은 원활하게 수정되어 작업을 끝냈지만, 일정이 빠듯하여 롤링 프로젝트의 특정 기능을 추가하는 작업은 하지 못한것이 아쉽다

현기)
- 처음에 정했던 폴더 구조에 대한 컨벤션이 제대로 지켜지지 않았다.

채연)
- 프로젝트를 진행할수록 맞춰나가야 할 디테일한 부분이 계속 나타나서 디테일을 완성하지 못한 듯해서 아쉬웠다.
- 일단 머지하기 바빠서 다른 사람의 코드를 충분히 검토하지 못한 것 같아 아쉽다.
 
 ### [Try]
근아)
- css 및 공통컴포넌트를 논의할 때 팀 피그마를 활용하는 팀이 있던데, 효율적이고 소통하기 좋아보였다. 다음 팀프로젝트때 사용해보면 좋을 것 같다.

우진)
- 개발 시작 전 개발단위에 대한 설정이 어려웠고, 개발단위를 작은 범위로 설정한 것 같아 다음엔 조금 더 큰 범위로 잡아서 진행해보고싶다.

성원)
- 다음 팀 프로젝트에서는 초반에 일정관리를 제대로 하여 요구사항 외의 추가 기능과 리팩토링 작업을 충분히 할 수 있으면 좋을거 같다

현기)
- 폴더의 구조와 파일의 분류 기준을 상세하게 정하면 좋을 거 같다.

채연)
- 다음 프로젝트에서는 티켓 단위를 명확히 정해 일정 관리하는 연습을 하는 것이 좋을 것 같다.

**Comment by mijin0928** [comment]
# 6팀

![기초프로젝트 1주차 회고록 2](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/94034865/0e799653-0824-487d-b0a3-751b0f0d9bfa)
